gc.c

```diff
@@ -621,17 +621,29 @@ typedef struct RVALUE {
             VALUE v3;
         } values;
     } as;
+
+    /* Start of RVALUE_OVERHEAD.
+     * Do not directly read these members from the RVALUE as they're located
+     * at the end of the slot (which may differ in size depending on the size
+     * pool). */
+#if RACTOR_CHECK_MODE
+    uint32_t _ractor_belonging_id;
+#endif
 #if GC_DEBUG
     const char *file;
     int line;
 #endif
 } RVALUE;
 
-#if GC_DEBUG
-STATIC_ASSERT(sizeof_rvalue, offsetof(RVALUE, file) == SIZEOF_VALUE * 5);
+#if RACTOR_CHECK_MODE
+# define RVALUE_OVERHEAD (sizeof(RVALUE) - offsetof(RVALUE, _ractor_belonging_id))
+#elif GC_DEBUG
+# define RVALUE_OVERHEAD (sizeof(RVALUE) - offsetof(RVALUE, file))
 #else
-STATIC_ASSERT(sizeof_rvalue, sizeof(RVALUE) == SIZEOF_VALUE * 5);
+# define RVALUE_OVERHEAD 0
```

Comment:
how about `RACTOR_CHECK_MODE && GC_DEBUG` case?

Reply:
If only `RACTOR_CHECK_MODE`, then `sizeof(RVALUE) == 48` and `offsetof(RVALUE, _ractor_belonging_id) == 40`, so `RVALUE_OVERHEAD == 8`.

If `RACTOR_CHECK_MODE && GC_DEBUG`, then `sizeof(RVALUE) == 64` and `offsetof(RVALUE, _ractor_belonging_id) == 40`, so `RVALUE_OVERHEAD == 24`.

So the first case (`sizeof(RVALUE) - offsetof(RVALUE, _ractor_belonging_id)`) works for `RACTOR_CHECK_MODE && GC_DEBUG`.